modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/DefaultSourceCallback.java

```diff
@@ -216,11 +216,9 @@ private void executeFlowTransactionally(SourceCallbackContext context, MessagePr
   }
 
   private void executeFlow(SourceCallbackContext context, MessageProcessContext messageProcessContext, Message message) {
-    messageProcessingManager.processMessage(new ModuleFlowProcessingTemplate(message,
-                                                                             listener,
-                                                                             completionHandlerFactory
-                                                                                 .createCompletionHandler(context),
-                                                                             messageProcessContext),
+    messageProcessingManager.processMessage(
```

Comment:
We should take a closer look at this and see if messageProcessingManager is really needed, or just add a layer of abstraction/complexity (different issue of course)